database/rrdset.c

```diff
@@ -1342,12 +1342,15 @@ void rrdset_done(RRDSET *st) {
     struct timeval now;
 
     now_realtime_timeval(&now);
-    rrdset_timed_done(st, now);
+    rrdset_timed_done(st, now, /* pending_rrdset_next = */ true);
 }
 
-void rrdset_timed_done(RRDSET *st, struct timeval now) {
+void rrdset_timed_done(RRDSET *st, struct timeval now, bool pending_rrdset_next) {
     if(unlikely(netdata_exit)) return;
 
+    if (pending_rrdset_next)
```

Comment:
This could do with a `likely()` wrapper.

Reply:
Better to leave this as it is. The plugin's parser will set this to false.

Reply:
OK!